### PR TITLE
[indexer-alt] Add benchmark command

### PR DESCRIPTION
**Pull request by lxfind**
## Description 

This PR adds a benchmark command to indexer-alt.
It would start the indexer using the provided parameters, start and stop based on the local ingestion data.
It then measures the time and TPS.
A few structural changes to indexer-alt code:
1. Moved lag and pruner options to a dedicated sequential pipeline config, since these params only make sense in a sequential pipeline setting.
2. Moved all the indexer start logic to a dedicated function, so that I can reuse it in the benchmark.
3. Added option to not bootstrap genesis if not needed.

## Test plan 

Run locally

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-indexer-alt/src/args.rs [hunk 1]
```diff
@@ -24,28 +24,8 @@ pub enum Command {
...
-            value_name = "SECONDS",
-            value_parser = |s: &str| s.parse().map(Duration::from_secs),
-        )]
-        pruner_delay: Duration,
-
-        /// Number of checkpoints to delay indexing summary tables for.
-        #[clap(long)]
-        consistent_range: Option<u64>,
+        #[command(flatten)]
+        sequential_pipeline_config: SequentialPipelineConfig,
```
Comment (amnn) [inline crates/sui-indexer-alt/src/args.rs:28]: nit: this is even more specific than for sequential pipelines -- let's call it a `ConsistencyConfig` or something?

File: crates/sui-indexer-alt/Cargo.toml [hunk 2]
```diff
@@ -42,6 +42,7 @@ sui-pg-temp-db.workspace = true
 sui-protocol-config.workspace = true
 sui-storage.workspace = true
 sui-types.workspace = true
+sui-synthetic-ingestion.workspace = true
```
Comment (amnn) [inline crates/sui-indexer-alt/Cargo.toml:45]: Seeing as we are making a fresh start -- is it worth making this an optional feature, guarded by a flag?

File: crates/sui-indexer-alt/src/lib.rs [hunk 3]
```diff
@@ -318,3 +339,83 @@ impl Indexer {
...
     }
 }
+
+pub async fn start_indexer(
+    indexer_config: IndexerConfig,
+    db_config: DbConfig,
+    sequential_pipeline_config: SequentialPipelineConfig,
+    // If true, the indexer will bootstrap from genesis.
+    // Otherwise it will skip the pipelines that rely on genesis data.
+    with_genesis: bool,
```
Comment (amnn) [inline crates/sui-indexer-alt/src/lib.rs:349]: Is it worth doing this the other way? I.e. infer whether we need to bootstrap genesis based on whether the pipelines we're about to run need it or not? Seems like that would be less error-prone.

File: crates/sui-indexer-alt/src/pipeline/sequential/mod.rs [hunk 4]
```diff
@@ -18,6 +18,7 @@ use super::{processor::processor, PipelineConfig, Processor, PIPELINE_BUFFER};
 use self::committer::committer;
 
 mod committer;
+pub mod config;
```
Comment (amnn) [inline crates/sui-indexer-alt/src/pipeline/sequential/mod.rs:21]: Spurious?

File: crates/sui-indexer-alt/src/main.rs [hunk 5]
```diff
@@ -1,30 +1,13 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use anyhow::{Context, Result};
+use anyhow::Result;
 use clap::Parser;
+use sui_indexer_alt::args::Args;
 use sui_indexer_alt::args::Command;
-use sui_indexer_alt::bootstrap::bootstrap;
+use sui_indexer_alt::benchmark::run_benchmark;
```
Comment (amnn) [inline crates/sui-indexer-alt/src/main.rs:8]: Don't think this module is in the PR.

**Review by amnn — COMMENTED** [review]
I think we're missing some files in the PR -- what's here seems reasonable, but I left some questions for your consideration! Thanks @lxfind .

File: crates/sui-indexer-alt/src/lib.rs (hunk 3 above)
Comment (lxfind) [reply crates/sui-indexer-alt/src/lib.rs:349]: Is there a better way to infer that other than hardcode the list of pipelines that need genesis somewhere?

File: crates/sui-indexer-alt/Cargo.toml [hunk 6]
```diff
@@ -42,8 +42,13 @@ sui-pg-temp-db.workspace = true
...
+sui-synthetic-ingestion = { workspace = true, optional = true }
 
 [dev-dependencies]
 rand.workspace = true
 tempfile.workspace = true
 wiremock.workspace = true
+
+[features]
+default = []
+benchmark = ["sui-synthetic-ingestion"]  # This will be used to enable benchmark mode
```
Comment (amnn) [inline crates/sui-indexer-alt/Cargo.toml:54]: This is mostly for my own education -- in some places, I've seen this written like it is here, and in other cases the feature name is prefixed with a "dep:" when it enables the dependency.

Is there a difference between the two?

File: crates/sui-indexer-alt/src/benchmark.rs [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+    #[command(flatten)]
+    pipeline_config: PipelineConfig,
+
+    /// Only run the following pipelines. If not provided, all pipelines will be run.
+    #[arg(long, action = clap::ArgAction::Append)]
+    pipeline: Vec<String>,
+
+    #[command(flatten)]
+    sequential_pipeline_config: ConsistencyConfig,
```
Comment (amnn) [inline crates/sui-indexer-alt/src/benchmark.rs:28]: nit: naming.
```suggestion
    consistency_config: ConsistencyConfig,
```

File: crates/sui-indexer-alt/src/pipeline/sequential/consistency_config.rs
Comment (amnn) [inline crates/sui-indexer-alt/src/pipeline/sequential/consistency_config.rs:1]: nit: I don't think this belongs inside `pipeline` -- it's specific to our indexer, while `pipeline` is part of the indexer framework. Let's keep it at the top-level for now.

**Review by amnn — APPROVED** [review]
Looks good! Just some small nits.

There's also the small thread about bootstrapping genesis -- as I was thinking about it, that discussion dove-tails with a separate discussion in the standalone synthetic benchmark generation logic where you get rid of checkpoint 0 -- is it worth simplifying things by keeping both? I.e. simulacrum always generates the genesis checkpoint, and we always bootstrap the genesis table, even during benchmarks?

File: crates/sui-indexer-alt/Cargo.toml (hunk 6 above)
Comment (lxfind) [reply crates/sui-indexer-alt/Cargo.toml:54]: Good question. I am not confused on the difference too.

**Pull request merged**